kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/outline/PsiLabelProvider.java

```diff
@@ -103,6 +113,8 @@ private String getPresentableElement(KtElement declaration) {
                     KtTypeReference typeReference = parameter.getTypeReference();
                     if (typeReference != null) {
                         text += typeReference.getText();
+                    } else {
+                        text += computeReturnType(function);
```

Comment:
Don't hurry, this type reference belongs to parameter, not return type :) 
You could check function like this:
`fun foo() = 123`, there is no return type in the Outline view